docs/apache-airflow/tutorial/taskflow.rst

```diff
@@ -365,7 +365,11 @@ You can apply the ``@task.sensor`` decorator to convert a regular Python functio
 BaseSensorOperator class. The Python function implements the poke logic and returns an instance of
 the ``PokeReturnValue`` class as the ``poke()`` method in the BaseSensorOperator does. The ``PokeReturnValue`` is
 a new feature in Airflow 2.3 that allows a sensor operator to push an XCom value as described in
-section "Having sensors return XOM values" of :doc:`apache-airflow-providers:howto/create-update-providers`.
+section "Having sensors return XCOM values" of :doc:`apache-airflow-providers:howto/create-update-providers`.
+
+Alternatively in cases where the sensor doesn't need to push XCOM values:  both ``poke()`` and the wrapped
+function can return a boolean-like value where ``True`` designates the sensor's operation as complete and
+``False`` designates the sensor's operation as incomplete.
```

Comment:
While this may work, I'm not sure this was intended? It isn't typed that way at least...

https://github.com/apache/airflow/blob/fbbd453d27fc9c6ac93f68de9e083ca684bec562/airflow/decorators/sensor.py#L59

Reply:
I believe on older airflows this was the expectation for sensors. Even the new object has __bool__ to maintain that compatibility. Therefore I think it is useful to note and document.

Reply:
Maybe for both worlds we update the return type to be either the new type or bool?

Reply:
It sounds like from the doc in the taskflow tutorial that bool should work, but we might have to test that?

Reply:
I'm going to go and check that... and look at the tests.. if it makes sense i'll update the return and come back.. stay tuned.

Reply:
Updated tests and return values to denote this being ok.